spacy/tokens/_retokenize.pyx

```diff
@@ -167,6 +167,26 @@ def _merge(Doc doc, int start, int end, attributes):
     cdef const LexemeC* lex = doc.vocab.get(doc.mem, new_orth)
     # House the new merged token where it starts
     cdef TokenC* token = &doc.c[start]
+    # Initially set attributes to attributes of span root
+    token.tag = doc.c[span.root.i].tag
+    token.pos = doc.c[span.root.i].pos
+    token.morph = doc.c[span.root.i].morph
+    token.ent_iob = doc.c[span.root.i].ent_iob
+    token.ent_type = doc.c[span.root.i].ent_type
+    merged_iob = token.ent_iob
+    # If span root is part of an entity, merged token is B-ENT
+    if token.ent_iob in (1, 3):
+        merged_iob = 3
+        # If start token is I-ENT and previous token is of the same
+        # type, then I-ENT (could check I-ENT from start to span root)
+        if doc.c[start].ent_iob == 1 \
+                and doc.c[start].ent_type == token.ent_type \
+                and doc.c[start - 1].ent_type == token.ent_type:
```

Comment:
Can `start` be 0 here? If so the `start-1` will introduce an out-of-bounds access.